ibis/backends/trino/registry.py

```diff
@@ -99,6 +102,48 @@ def _timestamp_truncate(t, op):
     return sa.func.date_trunc(precision, sa_arg)
 
 
+def _date_from_ymd(t, op):
+    ymdstr = sa.func.format(
+        '%04d-%02d-%02d',
+        t.translate(op.year),
+        t.translate(op.month),
+        t.translate(op.day),
+    )
+    return sa.func.date(ymdstr)
+
+
+def _time_from_hms(t, op):
+    timestr = sa.func.format(
+        '%02d:%02d:%02d',
+        t.translate(op.hours),
+        t.translate(op.minutes),
+        t.translate(op.seconds),
+    )
+    return sa.cast(timestr, sa.TIME)
+
+
+def _timestamp_from_ymdhms(t, op):
+    y, mo, d, h, m, s = (t.translate(x) if x is not None else None for x in op.args)
+    timestr = sa.func.format('%04d-%02d-%02d %02d:%02d:%02d', y, mo, d, h, m, s)
+    return sa.cast(timestr, sa.TIMESTAMP)
```

Comment:
```suggestion
    timestr = sa.func.format('%04d-%02d-%02dT%02d:%02d:%02d', y, mo, d, h, m, s)
    return sa.func.from_iso8601_timestamp(timestr)
```

This way, the code is more explicit about the intent.